### PR TITLE
Snackbar update

### PR DESCRIPTION
**Pull request by gumaerc**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes https://github.com/mitodl/open-discussions/issues/2784

#### What's this PR do?
Replaces our custom Snackbar component with the RMWC wrapped equivalent, but still tied to our redux state in the same way.  Our component has also been converted to a hook.

#### How should this be manually tested?
Go to the settings page and change your notification settings.  You should get a snackbar message confirming the change.

#### Screenshots (if appropriate)
Screen Shot 2020-05-04 at 4 49 16 PM
image

#### What GIF best describes this PR or how it makes you feel?
admiral ackbar

### CODE REVIEW

**Comment by aliceriot** [comment]
I think we might have some styling issues:

Screenshot from 2020-05-04 13-14-00

at some widths it appears that a div is overflowing it's container

**Comment by gumaerc** [comment]
> is it the right font size / type ?

I've been going off screenshots in an old PR because it's the best example I could find of it working:

image

The font looks similar, but maybe @aliceriot could weigh in?  I think positionally, the styles look better now.

**Comment by aliceriot** [comment]
yeah, before we just had a little black rectangle, it wasn't styled all that much tbh.

File: static/scss/snackbar.scss [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
     padding-top: 10px;
     padding-bottom: 10px;
   }
 
-  .mdc-snackbar__text {
+  .mdc-snackbar__surface {
+    background-color: $message-component-bg-color;
+    border-radius: $std-border-radius;
     margin: 5px 0;
     padding: 0;
```
Comment (aliceriot) [inline static/scss/snackbar.scss:21]:
```suggestion
    padding: 0;
    
    .mdc-snackbar__label {
      width: 100%;
      text-align: center;
    }
```

Comment (aliceriot) [reply static/scss/snackbar.scss:21]: quick suggestion that makes the label look like this:

Screenshot from 2020-05-05 10-42-55

**Comment by gumaerc** [comment]
@aliceriot this should be all set now

**Pull request merged**